fil_fungible_token/src/token/types.rs

```diff
@@ -140,50 +187,47 @@ pub struct AllowanceReturn {
     pub amount: TokenAmount,
 }
 
-impl Cbor for ChangeAllowanceParams {}
-impl Cbor for GetAllowanceParams {}
+impl Cbor for IncreaseAllowanceParams {}
+impl Cbor for DecreaseAllowanceParams {}
 impl Cbor for RevokeAllowanceParams {}
+impl Cbor for GetAllowanceParams {}
 impl Cbor for AllowanceReturn {}
 
-/// Burns an amount of token from an address
+/// Instruction to burn an amount of tokens
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct BurnParams {
-    pub owner: Address,
+    /// A non-negative amount to burn
     #[serde(with = "bigint_ser")]
     pub amount: TokenAmount,
-    pub data: RawBytes,
 }
 
+/// Instruction to burn an amount of tokens from another address
 #[derive(Serialize_tuple, Deserialize_tuple)]
-pub struct BurnReturn {
-    pub by: Address,
+pub struct BurnFromParams {
     pub owner: Address,
+    /// A non-negative amount to burn
     #[serde(with = "bigint_ser")]
-    pub burnt: TokenAmount,
-    #[serde(with = "bigint_ser")]
-    pub remaining_balance: TokenAmount,
+    pub amount: TokenAmount,
 }
 
-impl Cbor for BurnParams {}
-impl Cbor for BurnReturn {}
-
+/// The updated value after burning
 #[derive(Serialize_tuple, Deserialize_tuple)]
-pub struct TransferParams {
-    pub from: Address,
-    pub to: Address,
+pub struct BurnReturn {
+    /// New balance in the account after the successful burn
     #[serde(with = "bigint_ser")]
-    pub amount: TokenAmount,
-    pub data: RawBytes,
+    pub remaining_balance: TokenAmount,
 }
 
+/// The updated value after a delegated burn
 #[derive(Serialize_tuple, Deserialize_tuple)]
-pub struct TransferReturn {
-    pub from: Address,
-    pub to: Address,
-    pub by: Address,
+pub struct BurnFromReturn {
+    /// New balance in the account after the successful burn
     #[serde(with = "bigint_ser")]
-    pub amount: TokenAmount,
+    pub remaining_balance: TokenAmount,
+    /// New remaining allowance between the owner and operator (caller)
+    #[serde(with = "bigint_ser")]
+    pub remaining_allowance: TokenAmount,
```

Comment:
I'm ambivalent about the `remaining_` prefix on this one (so long as consistent with TransferFrom)